### PR TITLE
Added instructions on contributing

### PR DESCRIPTION
**Pull request by petevg**
Has some notes one the somewhat convulted process of modifying a
Bigtop-based charm, and submitting it to the right places in both the
Juju project and the Bigtop project.

@juju-solutions/bigdata

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## Contributing
+
+This charm has been built on top of the Apache Bigtop project. Its source code lives in the main Bigtop repository on Github: https://github.com/apache/bigtop
+
+Bug fixes, new features and other contributions are welcome, and encouraged! Since this is a joint project between Canonical Charmers and the Apache Bigtop community, it can be a little tricky to get the code submitted to all the right places, and +1ed by the right people, however.
```
Comment (johnsca) [inline CONTRIBUTING.md]: I think the steps outlined further down are fairly straightforward and it's not really all that "tricky."  Perhaps we could go more with something like:

> Bug fixes, new features, and other contributions are welcome and encouraged!  Contributing follows the [same process](https://cwiki.apache.org/confluence/display/BIGTOP/How+to+Contribute) as for the rest of the Bigtop project, but can be simplified by using the GitHub integration.  The simplified workflow is as follows:

**Comment by ktsakalozos** [comment]
I am bit sceptical on where should the instructions on contributing live. Here is why, as time goes by and we refine our workflow on submitting patches and charms upstream the contribution process might need to change. For example, there are talks of creating a branch for the work we are doing on charms, should the contributions on charms be against that branch? What if the contribution process changes in an unexpected way we cannot foresee at this point (gets more complicated or simplified). What would be the action items for us if the PR template format changes? We would have to re-promulgate all the charms to keep the contributions section consistent with reality. The point I am making it that we should give us the option to change the contribution process at will without the need to re-release everything.

So I would follow the BigTop paradigm, keep the details of the contribution instructions in a wiki and point to that wiki from the charms. Apart from the wiki link in the CONTRIBUTING.md we could put a general description on how this charm is built (bigtop, layers), contact details of the team.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+1. Make a ticket to track the work that you're doing in Bigtop's Jira: https://issues.apache.org/jira/browse/BIGTOP/
+2. Make a fork of the Bigtop project in your own Github account.
+3. You'll find code for this charm in ``bigtop-packages/src/charms/<charm-name>`` in your copy of the repo. Make your changes there.
+4. Submit your changes by creating a pull request from your fork to the main bigtop repository. Apache's Jira will automatically attach your pull request to your Jira ticket if you follow the formatting guidelines below.
+5. To make your charm publicly available for others to use, build it, and submit it to the [Charm Store](https://jujucharms.com/docs/2.0/authors-charm-store).
+
+Please feel free to ping us on freenode#juju, or email juju@lists.canonical.com if you have any questions. Happy coding!
+
+## Pull Request Formatting
```
Comment (ktsakalozos) [inline CONTRIBUTING.md]: I wouldn't put the template here instead I would point people to: https://cwiki.apache.org/confluence/display/BIGTOP/How+to+Contribute

Comment (johnsca) [inline CONTRIBUTING.md]: Unfortunately, that link doesn't actually document the automated GitHub integration, but I suppose we can submit a change to that.

**Comment by ktsakalozos** [comment]
Thank you Pete. It is a +1 from me.

**Pull request merged**